### PR TITLE
EVG-6439 use correct thread level for perf comparisons

### PR DESCRIPTION
**Pull request by john-m-liu**
- Expose the selected threads data (currently scoped to only the trend chart) by passing a closure which returns the data to the parent scope
- Use the selected threads data to filter both the baseline and hover sample data
- Refactor the HTML template a bit to have less inline computation

### CODE REVIEW

File: public/static/app/perf/TestSample.js [hunk 1]
```diff
@@ -19,28 +18,30 @@ mciModule.factory('TestSample', function() {
...
+      return this.sample.revision.substring(0, 7)
     }
 
     // Returns only the keys that have results stored in them
-    this.resultKeys = function(testName){
+    this.resultKeys = function (testName) {
       var testInfo = this.resultForTest(testName);
-      return _.pluck(_(testInfo.results).pairs().filter(function(x){return typeof(x[1]) == "object"}), 0)
+      return _.pluck(_(testInfo.results).pairs().filter(function (x) {
+        return typeof (x[1]) == "object"
```
Comment (jimoleary) [inline public/static/app/perf/TestSample.js]: Either version of this code will filter __undefined__ but not __null__:

```
typeof null
"object"
var x
testInfo = {results: {25: {AverageLatency: 3431223.721315061}, 30: null, 60: x, 120: 1}}
_.pluck(_(ttestInfo.results).pairs().filter(function (x) {return typeof (x[1]) == "object"}), 0)
["25", "30"]

// The following will do what I think you want:
_.chain(testInfo.results).pick(_.isObject).keys().value()
["25"]
```

Would it make sense to simplify it to the following:
```
return _.chain(testInfo.results).pick(_.isObject).keys().value()
```

File: public/static/app/perf/TestSample.js [hunk 2]
```diff
@@ -59,27 +62,30 @@ mciModule.factory('TestSample', function() {
...
     }
 
-    this.maxThroughputForTest = function(testName, metric){
-      if(!_.has(this._maxes, testName)){
-        var d = this.resultForTest(testName);
-        if(!d){
-          return;
-        }
-        this._maxes[testName] = _.max(
+    this.maxThroughputForTest = function (testName, metric, threadLevel) {
```
Comment (jimoleary) [inline public/static/app/perf/TestSample.js]: I suspect that this should return null or undefined for no value. And the calling code should handle null as no results.

The max value could actually be 0.

File: public/static/app/perf/TrendSamples.js [hunk 3]
```diff
@@ -103,32 +102,36 @@ mciModule.factory('TrendSamples', function() {
...
     }
 
-    this.tasksByCommitOrderByTestName = function(testName){
-       if(!(testName in this._tasksByName)){
-          this._tasksByName[testName] = _.sortBy(_.uniq(this.seriesByName[testName], function(x){return x.task_id}), "order")
-       }
-       return this._tasksByName[testName]
+    this.tasksByCommitOrderByTestName = function (testName) {
+      if (!(testName in this._tasksByName)) {
+        this._tasksByName[testName] = _.sortBy(_.uniq(this.seriesByName[testName], function (x) {
```
Comment (jimoleary) [inline public/static/app/perf/TrendSamples.js]:
```
_.chain(this.seriesByName[testName]).sortBy("order").uniq(true, x =>{return x.task_id}).value()
```

File: public/static/app/perf/TestSample.js (hunk 2 above)
Comment (john-m-liu) [reply public/static/app/perf/TestSample.js]: fixed

File: public/static/app/perf/TrendSamples.js (hunk 3 above)
Comment (john-m-liu) [reply public/static/app/perf/TrendSamples.js]: Sorry, this got changed by my formatter. I'd prefer to not change this logic unless there's a specific reason to

File: public/static/app/perf/TestSample.js (hunk 1 above)
Comment (john-m-liu) [reply public/static/app/perf/TestSample.js]: Sorry, this got changed by my formatter. I'm hesitant to change it at this time

**Pull request merged**